### PR TITLE
Add tooltip component

### PR DESCRIPTION
**Pull request by firthh**

### CODE REVIEW

File: src/components/Tooltip/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  render() {
+    const { position, small, trigger } = this.props;
+    const { active } = this.state;
+    const parentClassName = cx({
+      'us-tooltip': true,
+      'us-tooltip--active': active,
+      'us-tooltip--top': position === 'top',
+      'us-tooltip--bottom': position === 'bottom',
+      'us-tooltip--right': position === 'right',
+      'us-tooltip--left': position === 'left',
```
Comment (domtronn) [inline src/components/Tooltip/index.js]: This is a stylistic thing, but for this position stuff, you could do something like

```js
  // ...
  [`us-tooltip--${position}`]: !!position
  // ...
```

And then enumerate the different position types in the `PropTypes` as

```js
  position: PropTypes.oneOf([ 'left', 'right', 'bottom', 'top' ])
```

To me this seems more concise/explicit of the API _(don't have to update the `cx` call to add more position variants as classes)_ but I understand that it's perhaps trying to be _"clever"_ and probably makes it harder to read 🙂

**Comment by thomasbritton** [comment]
Yeah looks alright to me.

We have got an addition in Broadband that will scroll to the tooltip on mobile viewports because there can be times where you'll trigger the tooltip but the tooltip note will be out of the viewport due to the nature of how the tooltips are positioned on mobiles to be beneath the element.

You can see it in action if you click on a tooltip in our filters panel on our product table pages on mobile viewports

File: src/components/Tooltip/index.js (hunk 1 above)
Comment (firthh) [reply src/components/Tooltip/index.js]: I like that. I'll make those changes

**Comment by firthh** [comment]
@thomasbritton I've updated it to support that option. Unfortunately due to lack of support for `scrollIntoView` options it can only really go to the top or the bottom of the page. Is it worth adding something to check if the tooltip is already in view?

**Pull request merged**